### ISSUE TITLE
Infinite loop while refreshing the token

### ISSUE THREAD

**Issue by wilmveel** [AzureAD/azure-activedirectory-library-for-js#78]
Hello,

I'm implementing the adal js library for a mobile project project with cordova. Everething works fine but when the token expires and the application starts refreshing in the iframe it  will end up in an infinite loop. Two iframes are added one with the names adalRenewFrame and undefined. My configuration looks like this.

var localUri = window.location.href;
    var redirectUri = "http://redirect_uri";

    var config = {
        tenant: "xxxx",
        clientId: 'xxxx',
        endpoints: {
            "/api": "xxxx",
        },
        displayCall: function(url){
            var ref = window.open(url, '_blank', 'location=no');
            ref.addEventListener('loadstart', function(event) {
                if((event.url).indexOf(redirectUri) === 0) {
                    ref.close();
                    window.location.href = event.url.replace(redirectUri + "/", localUri);
                    window.location.reload();
                } 
            });
        },
        cacheLocation: "localStorage"
    };

    if(localUri.indexOf("http://") !== 0 && localUri.indexOf("https://") !== 0){
        config.redirectUri = redirectUri;
    }

    adalAuthenticationServiceProvider.init(config, $httpProvider);

**Comment by omercs** [comment]
Are you only using adal.js without angular modules?
You need handle the callback for normal js.
```
// handle the redirect at the same page for iframe and login redirect
    if (adal.isCallback(window.location.hash)) {
        adal.handleWindowCallback();
    }
```

**Comment by wilmveel** [comment]
Hej,

Tnx for your fast response. After debuging I found the problem. On the main page there was a controller loaded on the body which was doing a http call (see code snipt). This was causing a problem in the Iframe wich was opening an other iframe in the iframe.

I removed the call and everything is working fine now.o

```javascript
.controller("Contoller", function($log, $scope, $http){
`
	$http.get("/api/me")
	.success(function(data){
		$scope.firstName = data.GivenName;
		$scope.lastName = data.Surname;
	});
});
``````````

Other strange thing is that there are always two iframes loaded on the page. One with the correct id and one with id undefined.

````````````
<iframe id="adalIdTokenFrame".................

<iframe id="undefined"..............................

````````
